### ISSUE TITLE
Frontmatter indent

### ISSUE THREAD

**Issue by sugarshin** [remarkjs/remark-lint#169]
Hi,

I need detect mixed indent with tab in Frontmatter. Is there a way?

**Comment by wooorm** [comment]
Could you provide examples of ok and not ok input?

**Comment by sugarshin** [comment]
ok ❌ 

```yaml
---
user:
  name: sugarshin
	url: github.com/sugarshin
---
```

ok ✅ 

```yaml
---
user:
  name: sugarshin
  url: github.com/sugarshin
---
```

Do you understand?

**Comment by wooorm** [comment]
This depends on what front matter you use. [`remark-frontmatter`](https://github.com/remarkjs/remark-frontmatter) can help detect the front matter, but as so many different flavours exist (yaml, toml, and JSON), I don’t think the core of remark-lint should support this.

However, it is of course definitely possible to create [external rules](https://github.com/remarkjs/remark-lint#list-of-external-rules).

For example, to check if the YAML is valid, something like this may work (not tested):

```js
var rule = require('unified-lint-rule');
var visit = require('unist-util-visit');
var generated = require('unist-util-generated');
var yaml = require('js-yaml');

module.exports = rule('remark-lint:yaml', yaml);

function yaml(ast, file) {
  visit(ast, 'yaml', visitor);

  function visitor(node) {
    if (!generated(node)) {
      try {
        yaml.safeLoad(node.value);
      } catch (err) {
        file.message(err, node)
      }
    }
  }
}
```

I’m closing this in favour of https://github.com/remarkjs/ideas/issues/3. If someone would like to work on this, comment there!

**Comment by kachkaev** [comment]
Why does linter visit the yaml node that remark-frontmatter creates? Was not it supposed to be skipped?

**Comment by wooorm** [comment]
In the above example, it is created by `remark-frontmatter`, and checked by the above code?
